### PR TITLE
fix: return error if pending_tx failed

### PR DESCRIPTION
**Pull request by u2**

### CODE REVIEW

File: shared/src/chain_state.rs [hunk 1]
```diff
@@ -390,13 +390,14 @@ impl ChainState {
                             return Err(e);
                         }
                         tx_pool.update_statics_for_add_tx(tx_size, cycles);
-                    } else if self
-                        .pending_tx(&mut tx_pool, Some(cycles), tx_size, tx)
-                        .is_ok()
-                    {
+                        return Ok(cycles);
```
Comment (jjyr) [inline shared/src/chain_state.rs]: Can we use `else if let Ok(cycles) {....}` to eliminate `return`? it's more readability

Comment (u2) [reply shared/src/chain_state.rs]: It's on purpose. If using `else if`, there will be more nesting conditions, lead the code to complicated. Return result for every branch.

**Comment by u2** [comment]
bors r=zhangsoledad,jjyr

**Comment by u2** [comment]
fix the different_txs_with_same_input integration test in https://github.com/nervosnetwork/ckb/pull/1432/commits/08a2b18d325621596033a53ca94e6a2b3f38c562

File: test/src/specs/tx_pool/different_txs_with_same_input.rs [hunk 2]
```diff
@@ -19,15 +20,10 @@ impl Spec for DifferentTxsWithSameInput {
...
-        let output = tx2_temp
-            .outputs()
-            .as_reader()
-            .get(0)
-            .unwrap()
-            .to_entity()
-            .as_builder()
-            .capacity(capacity_bytes!(40_000).pack())
+        let output = CellOutputBuilder::default()
+            .capacity(capacity_bytes!(100).pack())
```
Comment (u2) [inline test/src/specs/tx_pool/different_txs_with_same_input.rs]: Before the output capacity was too high, the transaction returned error while RPC sending.

File: test/src/specs/tx_pool/different_txs_with_same_input.rs [hunk 3]
```diff
@@ -37,7 +33,7 @@ impl Spec for DifferentTxsWithSameInput {
             .send_transaction(tx1.clone().data().into());
         node0
             .rpc_client()
-            .send_transaction(tx1.clone().data().into());
+            .send_transaction(tx2.clone().data().into());
```
Comment (u2) [inline test/src/specs/tx_pool/different_txs_with_same_input.rs]: I think it was a mistake before.

**Comment by u2** [comment]
bors r=jjyr,keroro520,zhangsoledad

**Pull request merged**